src/software/pipeline/main_panoramaMerging.cpp

```diff
@@ -113,58 +114,308 @@ int aliceVision_main(int argc, char** argv)
+        {
+            clampHalf = true;
+        }
+    } 
 
-    bool first = true;
-    image::Image<image::RGBAfColor> panorama;
+
+    int panoramaWidth = 0;
+    int panoramaHeight = 0;
+    oiio::ParamValueList metadata;
 
     for (auto viewItem : sfmData.getViews())
     {
         IndexT viewId = viewItem.first;
         if(!sfmData.isPoseAndIntrinsicDefined(viewId))
             continue;
 
+        const std::string warpedPath = viewItem.second->getMetadata().at("AliceVision:warpedPath");
+
         // Get composited image path
-        const std::string imagePath = (fs::path(compositingFolder) / (std::to_string(viewId) + ".exr")).string();
-        
+        const std::string imagePath = (fs::path(compositingFolder) / (warpedPath + ".exr")).string();
+
         // Get offset
-        oiio::ParamValueList metadata = image::readImageMetadata(imagePath);
-        const int offsetX = metadata.find("AliceVision:offsetX")->get_int();
+        metadata = image::readImageMetadata(imagePath);
+        panoramaWidth = metadata.find("AliceVision:panoramaWidth")->get_int();
+        panoramaHeight = metadata.find("AliceVision:panoramaHeight")->get_int();
+        break;
+    }
+
+    int tileCountWidth = std::ceil(double(panoramaWidth) / double(tileSize));
+    int tileCountHeight = std::ceil(double(panoramaHeight) / double(tileSize));
+
+
+    std::map<std::pair<int, int>, IndexT> fullTiles;
+    for (auto viewItem : sfmData.getViews())
+    {
+        IndexT viewId = viewItem.first;
+        if(!sfmData.isPoseAndIntrinsicDefined(viewId))
+        {
+            continue;
+        }
+
+        const std::string warpedPath = viewItem.second->getMetadata().at("AliceVision:warpedPath");
+
+        // Get composited image path
+        const std::string imagePath = (fs::path(compositingFolder) / (warpedPath + ".exr")).string();
+
+        // Get offset
+        int width = 0;
+        int height = 0;
+        oiio::ParamValueList metadata = image::readImageMetadata(imagePath, width, height);
         const int offsetY = metadata.find("AliceVision:offsetY")->get_int();
-        const int panoramaWidth = metadata.find("AliceVision:panoramaWidth")->get_int();
-        const int panoramaHeight = metadata.find("AliceVision:panoramaHeight")->get_int();
 
-        if (first) 
+        int offsetX = metadata.find("AliceVision:offsetX")->get_int();
+        if (offsetX < 0)
         {
-            panorama = image::Image<image::RGBAfColor>(panoramaWidth, panoramaHeight, true, image::RGBAfColor(0.0f, 0.0f, 0.f, 0.0f));
-            first = false;
+            offsetX += panoramaWidth;
+        }
+    
+        int left = std::floor(double(offsetX) / double(tileSize));
+        int top = std::floor(double(offsetY) / double(tileSize));
+        int right = std::ceil(double(offsetX + width - 1) / double(tileSize));
+        int bottom = std::ceil(double(offsetY + height - 1) / double(tileSize));
```

Comment:
I think it makes sense to keep doing division in integers. Such upwards rounded division can be implemented like this:

```
int divideRoundUp(int a, int b)
{
    return (a + b - 1) / b;
}
```

Which could be used here as:

`int right = divideRoundUp(offsetX + width - 1, tileSize);`

Reply:
https://github.com/alicevision/AliceVision/pull/1254 introduces such function and fixes several bugs in division code (unrelated to this PR).